probability/distributions.cpp

```diff
@@ -0,0 +1,311 @@
+/**
+ * \file
+ * \brief Implementation of Discrete & Continuous Distributions
+ * \author [Tuan Ho](https://github.com/ttuanho)
+ * \note The implementation is based on [this
+ * cheatsheet](https://web.cs.elte.hu/~mesti/valszam/kepletek)
+ * \todo
+ * <ul>
+ *  <li>
+ *       Adding Gaussian distribution
+ *  </li>
+ *  <li>
+ *       Adding mean and variance function for each distribution
+ *  </li>
+ *  <li>
+ *       Adding cdf function of Normal distribution
+ *  </li>
+ * </ul>
+ */
+
+#include <cassert>
+#include <cmath>
+#include <iostream>
+#include <vector>
+
+#define _USE_MATH_DEFINES
+
+constexpr double E = 2.71828182845904523536;
```

Comment:
this is also available in `cmath` header as `M_E`.

C++20 has a dedicated STL class `std::number` containing precision specific implementations. But we are using C++11. 

Other standard constants: https://www.gnu.org/software/libc/manual/html_node/Mathematical-Constants.html

Reply:
turned out I didn't use this const but thanks for the heads up